### DISCUSSION TITLE
Hide parseCid logic from developer

### DISCUSSION THREAD

**Discussion by yondonfu** [livepeer/livepeer.js#127] [Ideas]
The current code example for IPFS playback at https://livepeerjs.org/examples/react/ipfs-playback requires the dev to import `parseCid` from `livepeer/media` which needs to be called and the value passed to the Player.

Got some feedback that it would nice to go one step further and have a way to hide `parseCid` from the dev by default so that they have a way to just input an IPFS URL into a component and the `parseCid` logic is auto-executed by the component internally without exposing that function to the dev. Maybe a wrapper component around the Player that just takes the IPFS URL?

**Comment by 0xcadams** [comment]
This is actually already the case - if you look at the content in that step, https://livepeerjs.org/examples/react/ipfs-playback#step-2-automatic-playback, it indicates:

> this is only used for UI purposes, to provide feedback if the user input is valid.

This is only used to ensure that the UI input is valid. We can tweak the example code, since this is technically just shown for UI purposes in the docs (to show that the input is valid/invalid).

**Reply by yondonfu** [reply]
Got it! The feedback was given from a dev that looked at the docs really quickly so it probably looked like the parseCid function was required for the player to work at all, but in reality it is just a convenience function so that the UI input can immediately validate and show any problems with the input. In that case, the current setup seems fine for now 👍 .
